node/full.go

```diff
@@ -410,19 +410,19 @@ func (n *FullNode) GetGenesisChunks() ([]string, error) {
 // OnStop is a part of Service interface.
 func (n *FullNode) OnStop() {
 	n.Logger.Info("halting full node...")
-	n.cancel()
-	n.threadManager.Wait()
 	n.Logger.Info("shutting down full node sub services...")
-	err := n.p2pClient.Close()
-	err = errors.Join(
-		err,
+	err := errors.Join(
+		n.p2pClient.Close(),
 		n.hSyncService.Stop(),
 		n.bSyncService.Stop(),
 		n.IndexerService.Stop(),
 	)
 	if n.prometheusSrv != nil {
 		err = errors.Join(err, n.prometheusSrv.Shutdown(n.ctx))
 	}
+	n.cancel()
+	n.threadManager.Wait()
```

Comment:
can we explain why the context cancel needs to be moved down here? 
It isn't clear why the context should be cancelled after calling stop on the sub services. 
Same comment from above, this learning should be captured in comments here so that we don't accidentally undo this in the future.